docs/architecture.rst

```diff
@@ -1,66 +1,167 @@
+* A :class:`.YHalfCube` represents Y-basis initialization and measurements.
+* :class:`.ZXCube` defines cubes with only X or Z basis boundaries.
 
-TODO
 
-``tqec.circuit``
+:mod:`.compile`
 ----------------
 
-TODO
+Responsible for translations between internal representations.
+
+:class:`.BlockGraph` from the functionality in :ref:`interop_ref` is further translated into a compiled graph through :func:`.compile_block_graph`.
+
+.. mermaid::
+    :align: center
+    :config: {"theme": "base", "darkMode": "true"}
+
+    %%{
+    init: {
+        'theme': 'base',
+        'themeVariables': {
+        'primaryColor': '#AFEEEE',
+        'lineColor': '#6495ED'
+        }
+    }
+    }%%
+
+    graph
+        A[BlockGraph] --> B[compiled graph] --> C[ScheduledCircuit]--> D[stim.Circuit]
+
+Multiple block builder protocols defined in ``tqec/compile/observables/builder.py`` will take the high-level structure of a block to
+templates and plaquettes i.e. a fully annotated ``stim.Circuit``.
+
+:mod:`.templates`
+^^^^^^^^^^^^^^^^^
+
+Generates an array of numbers in a representation of a 2-dimensional, scalable, arrangement of plaquettes. This allows us to describe a
+circuit that can scale the desired code distance.
+
+
+:mod:`.plaquette`
+^^^^^^^^^^^^^^^^^
+
+A plaquette is the representation of a local quantum circuit. A surface code patch implements one layer in time or one round of the surface code.
+
+This module maps from the numbered templates to some plaquettes that implement small local circuits to measure a stabilizer as a :class:`.ScheduledCircuit` instance.
 
-``tqec.noise_models``
----------------------
 
-TODO
+:mod:`.circuit`
+^^^^^^^^^^^^^^^
 
-``tqec.simulation``
+Implementation of :class:`.ScheduledCircuit`, a quantum circuit representation in tqec, where each and every gate of a regular quantum circuit is associated with the time of execution.
+
+:class:`.NoiseModel`
+--------------------
+
+.. note::
+
+    The code for this module was modified from the code for :cite:`Gidney_inplace_access_2024`.
+
+This module implements the following noise models for ``Stim`` simulations:
+
+#. **Superconducting Inspired Circuit Error Model (SI1000)**: A modified version of the noise model introduced in :cite:`Gidney_si1000_2021` which represents the noise on Google's superconducting quantum chip.
+
+    In :meth:`.si1000`:
+
+    * Depolarizing noise on measured qubits from the noise modeil in :cite:`Gidney_si1000_2021` has been removed because ``tqec`` measurements are immediately followed by resets.
+
+    * The measurement result is probabilistically flipped instead of the input qubit.
+
+#. **Uniform Depolarizing Noise**: Single qubit depolarizing noise is uniformly applied to both single qubit and two qubit Clifford gates.
+
+    In :meth:`.uniform_depolarizing`:
+
+    * The result of dissipative gates is probabilistically bit or phase flipped.
+
+    * Result of non-demolition measurements is flipped instead of the input qubit.
```

Comment:
Now in `tqec.utils`

Reply:
Yes, if you click on the section heading or the noise methods, it takes you to `tqec.utils.NoiseModel`. Do you still want me to mention this is in `tqec.utils`?